### PR TITLE
Introduce a filter that short-circuits the avatar URL generation

### PR DESCRIPTION
**Pull request by johnbillion**
### Description of the Change

When a dynamic image resizing service (such as Tachyon or Photon) is in use, Simple Local Avatars provides no way for the URL of an avatar of a given size to be short-circuited so as to avoid its own dynamic resizing that happens via a `WP_Image_Editor` instance.

This change introduces a filter to short-circuit the URL for an avatar (once it's been determined that a "full" size exists), allowing a dynamic image resizing service to return its own scaled version of the user's image.

### How to test the Change

Here's a simplified example of how this filter could be used on a site that uses Tachyon:

```php
add_filter(
	'pre_simple_local_avatar_url',
	function( $url, $user_id, $size, $local_avatars ) {
		return tachyon_url( $local_avatars['full'], [ 'resize' => "{$size},{$size}" ] );
	},
	10,
	4
);
```

### Changelog Entry

> Added - `pre_simple_local_avatar_url` filter to allow an avatar image to be short-circuited before Simple Local Avatars processes it.

### Credits

@johnbillion

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by jeffpaul** [comment]
Thanks @johnbillion, that usage makes sense to me.  Tagging @faisal-alvi for code review/testing to see if there are any concerns here before moving towards merge/release.

**Comment by peterwilsoncc** [comment]
@faisal-alvi I have a copy of Tachyon set up locally if you need guidance on getting it going/want me to test.

**Comment by faisal-alvi** [comment]
@peterwilsoncc it would be great if you test + code review this as I do not have Tachyon set up locally. Thanks! ❤️

File: includes/class-simple-local-avatars.php
Comment (peterwilsoncc) [inline includes/class-simple-local-avatars.php]: The function is considered "safe" -- WP handles double escaping.

But hold off on actioning this given the comment above.

```suggestion
			return esc_url( $url );
```

**Review by peterwilsoncc — COMMENTED** [review]
This tests well and works as described.

I've just raised a discussion internally as to whether the plugin should switch to using `wp_get_attachment_image_src()[0]` and storing dynamically resized images in the image post meta. That will allow the plugin to just work with Tachyon, Photon and other services using the `image_downsize` filter to dynamically resize images.

File: includes/class-simple-local-avatars.php
Comment (johnbillion) [reply includes/class-simple-local-avatars.php]: I think that would be to be esc_url_raw anyway

**Review by peterwilsoncc — COMMENTED** [review]
@johnbillion We've considered the options and decided to put this in for now rather than switch to the proper media functions. The team needs to do a little more investigating of the implications for multisite installations using shared avatars. 

> I think that would be to be esc_url_raw anyway

Yeah, it probably should be but we use `esc_url()` elsewhere in the function so it's intended for printing rather than getting.. It's probably too late to make the change now.

**Review by peterwilsoncc — APPROVED** [review]

**Pull request merged**
